pkg/rakshctl/cmd/app/create_test.go

```diff
@@ -21,66 +21,39 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-func TestInsertVaultSecret(t *testing.T) {
-	testVaultSecretName := "test-vault-secret"
-	expectSecrets := []corev1.EnvVar{
-		{
-			Name: "SC_VAULT_ADDR",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: testVaultSecretName,
-					},
-					Key: "vaultAdd",
-				},
-			},
-		},
-		{
-			Name: "SC_VAULT_TOKEN",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: testVaultSecretName,
-					},
-					Key: "vaultToken",
-				},
-			},
-		},
-		{
-			Name: "SC_VAULT_SECRET",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: testVaultSecretName,
-					},
-					Key: "secretName",
-				},
-			},
-		},
-		{
-			Name: "SC_VAULT_SYMM_KEY",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: testVaultSecretName,
-					},
-					Key: "keyName",
-				},
+func TestMountRakshSecrets(t *testing.T) {
+	testRakshSecretName := "test-raksh-secret"
+	testVolumeName := "secure-volume-raksh"
+
+	expectedVolume := corev1.Volume{
+		Name: testVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: testRakshSecretName,
 			},
 		},
 	}
+
+	expectedVolmount := corev1.VolumeMount{
+		Name:      testVolumeName,
+		ReadOnly:  true,
+		MountPath: "/etc/raksh-secrets",
+	}
+
 	testpod := &corev1.PodSpec{
 		Containers: []corev1.Container{
 			{Name: "c1"},
-			{Name: "c2"},
```

Comment:
I'm wondering why is this removed? is it because of that multi-container deployment issue?

Reply:
yes.. want to add test for multi-container separately.. is it ok ?

Reply:
👍